### DISCUSSION TITLE
Playing http://icecast.omroep.nl/radio1-bb-mp3 times out; for DRM maybe ?

### DISCUSSION THREAD

**Discussion by PeterDemmer** [pschatzmann/arduino-audio-tools#665] [Q&A] [answered]
When playing this stream, after connection setup, no audio data are received and the connection times out.
With other streams, everything works fine.
I noticed that when playing this stream in a browser, usage of DRM content needs to be confirmed.
Anyone else having problems with this stream ? 
Any additional configuration required maybe ?

**Comment by pschatzmann** [comment]
Not sure if this helps, but did you try with an ICYStream ? 
After all this is Icecast...

**Reply by PeterDemmer** [reply]
Thanks, Phil.

I made the following changes to layer-url-i2s:
// URLStream urlStream(wifi, password);
ICYStream icyStream(wifi, password);
//AudioSourceURL source(urlStream, urls, "audio/mp3");
AudioSourceURL source(icyStream, urls, "audio/mp3");

Stream hangs at:
12:57:47.159 -> [I] HttpRequest.h : 228 - Request written ... waiting for reply
12:57:47.159 -> [I] HttpHeader.h : 225 - HttpHeader::read
12:57:47.159 -> [W] HttpHeader.h : 232 - Waiting for data...
12:58:08.170 -> [W] HttpLineReader.h : 27 - HttpLineReader readlnInternal->no Data
12:58:08.170 -> [I] HttpHeader.h : 162 - HttpHeader::readLine -> 
12:58:08.170 -> [I] AudioCopy.h : 71 - buffer_size=1024
12:58:08.170 -> [I] AudioPlayer.h : 326 - setVolume(0.481321)
12:58:08.170 -> [I] VolumeStream.h : 183 - setVolume: 0.480000
12:58:08.170 -> [I] VolumeStream.h : 183 - setVolume: 0.480000
12:58:28.175 -> [W] AudioPlayer.h : 445 - -> timeout - moving by 1
and proceeds to next 

So nothing changed with this adjustment.

Regards, Peter

**Reply by pschatzmann** [reply]
So maybe you might figure out what's going on if you set the log level to debug. There you can check what you really receive

**Reply by PeterDemmer** [reply]
Thanks, Phil.

I increased the loglevel to debug.

A working stream replies like this:
[I] HttpHeader.h : 162 - HttpHeader::readLine -> 'HTTP/1.1 200 OK'

and a non-working stream sends nothing:
17:01:39.716 -> [I] HttpRequest.h : 230 - Request written ... waiting for reply
17:01:39.749 -> [I] HttpHeader.h : 225 - HttpHeader::read
17:01:39.749 -> [W] HttpHeader.h : 232 - Waiting for data...
17:01:59.233 -> [D] HttpLineReader.h : 20 - HttpLineReader readlnInternal
17:02:01.256 -> [W] HttpLineReader.h : 27 - HttpLineReader readlnInternal->no Data
17:02:01.256 -> [I] HttpHeader.h : 162 - HttpHeader::readLine -> ''
17:02:01.256 -> [D] HttpHeader.h : 455 - HttpReplyHeader::parse1stLine: ''

**Reply by pschatzmann** [reply]
Right: this will not work if the request just returns an empty line instead of a valid HTTP status line

**Reply by PeterDemmer** [reply]
Hi Phil,

Thanks.

I found a way to fix this:
In libraries/arduino-audio-tools/src/AudioHttp/HttpRequest.h:
222c222
<             if (len>0){
---
>             if (true || len>0){

The log for https://icecast.omroep.nl/radio1-bb-mp now reads:
13:40:44.225 -> [I] HttpHeader.h : 300 -  -> <CR LF> 
13:40:44.225 -> [I] HttpRequest.h : 223 - Writing data: -1 bytes
13:40:44.285 -> [I] HttpRequest.h : 228 - Request written ... waiting for reply
13:40:44.285 -> [I] HttpHeader.h : 225 - HttpHeader::read
13:40:44.325 -> [W] HttpHeader.h : 232 - Waiting for data...
13:40:44.814 -> [I] HttpHeader.h : 162 - HttpHeader::readLine -> HTTP/1.1 200 OK
but writing -1 bytes seems to work fine.

**Reply by pschatzmann** [reply]
I am confused about your proposal. Can you be a little bit more specific or better submit a pull request.
In HttpRequest.h there is no such line...

**Reply by PeterDemmer** [reply, marked as answer]
Hi Phil,
I was using code dated 28 Feb., hence the confusion.
Sorry about that.
Regards,
Peter
